core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java

```diff
@@ -198,30 +200,34 @@ public void testOneOwnerSuspectedNoFilter(Method m) throws ExecutionException, I
       cache(1).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release1), 0);
       cache(2).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release2), 0);
 
-      org.infinispan.remoting.transport.Address address1 = cache(1).getCacheManager().getAddress();
-      org.infinispan.remoting.transport.Address address2 = cache(2).getCacheManager().getAddress();
-      List<org.infinispan.remoting.transport.Address> owners = Arrays.asList(address1, address2);
+      Address address1 = address(1);
+      Address address2 = address(2);
+      List<Address> owners = Arrays.asList(address1, address2);
 
       ClusteredGetCommand clusteredGet = new ClusteredGetCommand(key, ByteString.fromString(cache(0).getName()), 0, AnyEquivalence.getInstance());
       final int timeout = 15;
       RpcOptions rpcOptions = new RpcOptions(timeout, TimeUnit.SECONDS, null, ResponseMode.WAIT_FOR_VALID_RESPONSE, DeliverOrder.NONE);
 
-      CompletableFuture<Map<org.infinispan.remoting.transport.Address, Response>> future = cache(0).getAdvancedCache().getRpcManager().invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
+      RpcManager rpcManager = cache(0).getAdvancedCache().getRpcManager();
+      CompletableFuture<Map<Address, Response>> future = rpcManager.invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
 
       assertTrue(arrival.await(10, TimeUnit.SECONDS));
 
       installNewView(cache(0), cache(0), cache(1));
 
+      // RequestCorrelator processes the view asynchronously, so we need to wait a bit for node 2 to be suspected
+      Thread.sleep(100);
```

Comment:
I thought about subclassing `CommandAwareRpcDispatcher` and `CustomRequestCorrelator` to make `receiveView` synchronous, but I decided that it wasn't worth the effort.